Engine/src/Engine/Memory/MemoryManager.cpp

```diff
@@ -5,30 +5,30 @@ const MemoryManager::Size MemoryManager::BYTE = 1;
 const MemoryManager::Size MemoryManager::KB = 1024 * BYTE;
 const MemoryManager::Size MemoryManager::MB = 1024 * KB;
 const MemoryManager::Size MemoryManager::GB = 1024 * MB;
+
 MemoryManager* MemoryManager::instance;
-static MemoryManager g_instance(2 * MemoryManager::GB,
-	1 * MemoryManager::GB,
-	8 * MemoryManager::BYTE);
-MemoryManager::MemoryManager(Size mem_size,
-	Size stack_size, 
-	Size pool_chunk_size) :
-	m_stackAlloc(mem_size),
-	m_poolAlloc(),
-	m_persistentAlloc(),
-	m_total_size(mem_size)
-{
-	m_persistentAlloc.Init(stack_size,m_stackAlloc.Alloc(stack_size, 8));
-	m_poolAlloc.Init(pool_chunk_size,
-		m_stackAlloc.GetRemainingSize() / pool_chunk_size,
-		m_stackAlloc.GetRemainingSize() / pool_chunk_size,
-		m_stackAlloc.Alloc(m_stackAlloc.GetRemainingSize(), 8));
+static MemoryManager g_instance;
 
-	ENGINE_ASSERT_MSG(!instance, "MemoryManager instance already exists");
+MemoryManager::MemoryManager() :
+	m_persistentAllocator(2 * MemoryManager::GB),
+	//m_poolAlloc(),
+	m_total_size(2 * MemoryManager::GB)
+{
+	//ENGINE_ASSERT_MSG(instance != nullptr, "MemoryManager instance already exists");
+	if (instance != nullptr)
+	{
+		throw std::bad_alloc{};
+		std::cout << "MemoryManager instance already exists" << std::endl;
+		return;
```

Comment:
will this lines every be hit since you throw already?

Reply:
updated to print it out before the throw